src/main/client/featureToggleClientFactory.ts

```diff
@@ -0,0 +1,12 @@
+import { CoreOptions, RequestAPI } from 'request'
+import { RequestPromise } from 'request-promise-native'
+
+import { FeatureToggleClient } from './FeatureToggleClient'
+
+export class FeatureToggleClientFactory {
+
+  async create<T> (featureToggleUri: string,
```

Comment:
Please remove async and make it return `FeatureToggleClient`. There is nothing that would require it to be async.